File: src/main/java/io/realworld/api/ProfileApiImpl.java
package io.realworld.api;

import io.realworld.model.GetProfileByUsername200Response;

import io.vertx.core.Future;
import io.vertx.ext.web.handler.HttpException;

// Implement this class

public class ProfileApiImpl implements ProfileApi {
    public Future<ApiResponse<GetProfileByUsername200Response>> followUserByUsername(String username) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<GetProfileByUsername200Response>> getProfileByUsername(String username) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<GetProfileByUsername200Response>> unfollowUserByUsername(String username) {
        return Future.failedFuture(new HttpException(501));
    }

}

File: src/main/kotlin/io/realworld/api/ProfileApiImpl.kt
package io.realworld.api

import io.realworld.model.response.ApiResponse
import io.realworld.model.response.GetProfileByUsername200Response
import io.vertx.core.Future
import io.vertx.ext.web.handler.HttpException

class ProfileApiImpl : ProfileApi {
  override fun followUserByUsername(username: String): Future<ApiResponse<GetProfileByUsername200Response>> {
    return Future.failedFuture(HttpException(501))
  }

  override fun getProfileByUsername(username: String): Future<ApiResponse<GetProfileByUsername200Response>> {
    return Future.failedFuture(HttpException(501))
  }

  override fun unfollowUserByUsername(username: String): Future<ApiResponse<GetProfileByUsername200Response>> {
    return Future.failedFuture(HttpException(501))
  }
}
